src/bitmessagemain.py

```
@@ -233,66 +259,82 @@ def start(self):
+        if daemon and BMConfigParser().safeGet(
+                "bitmessagesettings", "smtpdeliver", ''):
             smtpDeliveryThread = smtpDeliver()
             smtpDeliveryThread.start()
 
         # SMTP daemon thread
-        if daemon and BMConfigParser().safeGetBoolean("bitmessagesettings", "smtpd"):
+        if daemon and BMConfigParser().safeGetBoolean(
+                "bitmessagesettings", "smtpd"):
             smtpServerThread = smtpServer()
             smtpServerThread.start()
 
         # Start the thread that calculates POWs
         objectProcessorThread = objectProcessor()
-        objectProcessorThread.daemon = False  # DON'T close the main program even the thread remains. This thread checks the shutdown variable after processing each object.
+        # DON'T close the main program even the thread remains.
+        # This thread checks the shutdown variable after processing
+        # each object.
+        objectProcessorThread.daemon = False
         objectProcessorThread.start()
 
         # Start the cleanerThread
         singleCleanerThread = singleCleaner()
-        singleCleanerThread.daemon = True  # close the main program even if there are threads left
+        # close the main program even if there are threads left
+        singleCleanerThread.daemon = True
         singleCleanerThread.start()
 
         shared.reloadMyAddressHashes()
         shared.reloadBroadcastSendersForWhichImWatching()
 
-        if BMConfigParser().safeGetBoolean('bitmessagesettings', 'apienabled'):
-            try:
-                apiNotifyPath = BMConfigParser().get(
-                    'bitmessagesettings', 'apinotifypath')
-            except:
-                apiNotifyPath = ''
-            if apiNotifyPath != '':
-                with shared.printLock:
-                    print('Trying to call', apiNotifyPath)
+        singleAPIThread = singleAPI()
+        singleAPIThread.daemon = True
+        singleAPIThread.start()
```

It should only run when the api is enabled.

---

I moved that check into `singleAPI.__init__`

---

For configuring the API yes, but not for starting the thread.

---

It will not start if `apienaled = true` not set in config, look at line 200 and `_getApiAddress` (re)definition

---

This is a little non-obvious. But it will still start the thread, even if it quits right away. Perhaps the init should raise an exception and then these 3 lines be wrapped in try/except instead.

---

Thread will be stopped then because [serve_forever](/Bitmessage/PyBitmessage/blob/master/src/api.py#L61) not called

---

What is the point of starting it when it quits right away? Isn't it better not to start it in the first place?